lib/cn/kvset.c

```diff
@@ -2366,11 +2366,21 @@ kblk_start_read(struct kvset_iterator *iter, struct kblk_reader *kr, enum read_t
         /* starting a new kblock */
         kblk = &iter->ks->ks_kblks[kr->kr_next_kblk_idx];
 
-        wbt = read_type == READ_WBT ? &kblk->kb_wbt_desc : &iter->ks->ks_hblk.kh_ptree_desc;
+        switch (read_type) {
+        case READ_WBT:
+            wbt = &kblk->kb_wbt_desc;
+            kr->kr_mbid = kblk->kb_kblk.bk_blkid;
+            kr->kr_next_kblk_idx++;
+            break;
+        case READ_PT:
+            wbt = &iter->ks->ks_hblk.kh_ptree_desc;
+            kr->kr_mbid = iter->ks->ks_hblk.kh_hblk.bk_blkid;
+            /* Reuse same kblock for next iteration since it went unused */
```

Comment:
Usually better for comments to precede the code they are referencing, no?

Reply:
I put the comment in the spot where the iteration of the next kblock index happened in the previous block.

Reply:
I'm just saying it's unnatural for a comment to reference the code above the comment unless it addressess it directly.  Typically a comment refers to the code that follows the comment.